addons/website_sale_digital/models/product.py

```diff
@@ -59,9 +60,27 @@ def action_open_attachments(self):
             'type': 'ir.actions.act_window',
             'view_mode': 'kanban,form',
             'view_type': 'form',
-            'context': "{'default_res_model': '%s','default_res_id': %d, 'default_product_downloadable': True}" % (self._name, self.id),
+            'context': {'default_res_model': self._name, 'default_res_id': self.id, 'default_product_downloadable': True, 'default_name': self.product_tmpl_id.name},
             'help': """
                 <p class="oe_view_nocontent_create">Click on create to add attachments for this digital product.</p>
                 <p>The attached files are the ones that will be purchased and sent to the customer.</p>
                 """,
         }
+
+    @api.multi
+    def get_digital_attachment(self):
```

Comment:
I don't think it can be much simplified or search less, the fact that you need to link the attachment to a `product.product` ID, makes it annoying when you find an attachment from its template.

If you do it in one search, you'll have to change the `res_id` for every `res_model == product.template` to know to which `product.product` it's linked (needed cause `sale.order.line` is linked to `product.product`).